transport.go

```diff
@@ -68,15 +67,14 @@ func NewTransport(c net.Conn, l LogFactory, wef WrapErrorFunc) Transporter {
 		cdec:             cdec,
 		log:              log,
 		wrapError:        wef,
-		encodeCh:         make(chan []byte),
-		encodeResultCh:   make(chan error),
+		encodeCh:         make(chan encodingBundle, 1),
```

Comment:
I don't know if buffering buys us anything here, actually. Call and Notify both block on errCh (and ctx.Done) so even if we had a long buffer, those would still block until their encode finishes. It would buy us strict ordering (for the buffer length) but I don't know if that's worth it.

On the other hand, in the future, we could imagine passing in a context to Encode() and having it select on ctx.Done and the send on encodeCh, and so if you cancel an RPC that hasn't been sent yet, it would avoid even talking to the server. That seems to be an argument for keeping encodeCh unbuffered.